src/test/util_string_tests.cpp

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2024-present The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <util/string.h>
+
+#include <boost/test/unit_test.hpp>
+
+using namespace util;
+
+BOOST_AUTO_TEST_SUITE(util_string_tests)
+
+BOOST_AUTO_TEST_CASE(ConstevalFormatString_NumSpec)
+{
+    // Compile-time sanity checks
+    static_assert([] {
+        ConstevalFormatString<0>::Detail_CheckNumFormatSpecifiers("");
+        ConstevalFormatString<0>::Detail_CheckNumFormatSpecifiers("%%");
+        ConstevalFormatString<1>::Detail_CheckNumFormatSpecifiers("%_");
+        ConstevalFormatString<0>::Detail_CheckNumFormatSpecifiers("%%_");
+        ConstevalFormatString<0>::Detail_CheckNumFormatSpecifiers("_%%");
+        ConstevalFormatString<1>::Detail_CheckNumFormatSpecifiers("%%%_");
+        ConstevalFormatString<1>::Detail_CheckNumFormatSpecifiers("%_%%");
+        return true; // All checks above compiled and passed
+    }());
+
+    // Negative checks at runtime
+    using ErrType = const char*;
+
+    struct CheckThrowNumSpec {
+        bool operator()(const ErrType& str) const { return std::string_view{str} == "Format specifier count must match the argument count!"; }
+    };
```

Comment:
Instead of a functor I think we can simplify this by using a lambda instead.
And we can also simplify the comparison by using string_literals, i.e.:
```C++
#include <string>
using namespace std::string_literals;
...
    auto check_throw_num_spec = [](ErrType str) { return str == "Format specifier count must match the argument count!"s; };
    BOOST_CHECK_EXCEPTION(ConstevalFormatString<1>::Detail_CheckNumFormatSpecifiers(""), ErrType, check_throw_num_spec);
...
```

Reply:
What's the advantage of the `string_view` version?

Reply:
ok, that's fair, thanks.